### PR TITLE
Enable/Disable Seeding during Runtime

### PR DESCRIPTION
**Pull request by gitpubber**
Closes https://github.com/anacrolix/torrent/issues/479

### CODE REVIEW

**Comment by gitpubber** [comment]
@anacrolix

**Comment by anacrolix** [comment]
Would you like to add some tests? Perhaps a test that it works to switch seeding on after initially starting off.

**Comment by gitpubber** [comment]
@anacrolix , i have written tests

**Comment by gitpubber** [comment]
@anacrolix  , I have also made checking whether it is seeding , mutex protected . I believe there are no dataraces here and code is fine.

File: torrent.go [hunk 1]
```diff
@@ -1444,7 +1444,7 @@ func (t *Torrent) seeding() bool {
 	if cl.config.NoUpload {
 		return false
 	}
-	if !cl.config.Seed {
+	if !cl.IsSeeding() {
```
Comment (anacrolix) [inline torrent.go:1447]: I don't know if it's okay to call a method here that obtains a read lock on the Client.

Comment (gitpubber) [reply torrent.go:1447]: Why not ? Does it acquire recursive read lock here ? 

Oh, Can we have seperate mutex lock for Seeding ?

**Comment by gitpubber** [comment]
@anacrolix  I think it must be correct now.

File: torrent.go (hunk 1 above)
Comment (gitpubber) [reply torrent.go:1447]: @anacrolix  I think next commit solves it

Comment (anacrolix) [reply torrent.go:1447]: I don't think a separate mutex is the requirement, it's recursively locking it.

**Comment by anacrolix** [comment]
I'm not sure I want to merge this yet, I haven't got enough testing, and I'm not sure how I want to expose this.

File: torrent.go (hunk 1 above)
Comment (gitpubber) [reply torrent.go:1447]: @anacrolix oh,ok. what might be optimal solution for this?

**Comment by gitpubber** [comment]
@anacrolix No problem. this means that Seed can't be changed during runtime. I wish that you come up with some solution so that Seed variable can be changed during runtime

**Pull request closed without merging**